packages/hurl_core/src/richtext/mod.rs

```diff
@@ -0,0 +1,119 @@
+/*
+ * Hurl (https://hurl.dev)
+ * Copyright (C) 2024 Orange
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+use colored::Colorize;
+
+/// An String with color and style
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[allow(unused)]
+struct RichText {
+    tokens: Vec<Token>,
+}
+
+#[allow(unused)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct Token {
+    content: String,
+    fg: Option<Color>,
+    bold: bool,
+}
+
+#[allow(unused)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+enum Color {
+    Red,
+    Green,
+    Blue,
+}
+
+#[allow(unused)]
+impl RichText {
+    pub fn new() -> RichText {
+        RichText { tokens: vec![] }
+    }
+
+    pub fn add(&mut self, content: &str, fg: Option<Color>, bold: bool) {
+        let token = Token::new(content, fg, bold);
+        self.tokens.push(token);
+    }
+
+    pub fn add_text(&mut self, content: &str) {
```

Comment:
We could also make a "fluid" api: instead of `add_text`, this one:

```rust
    pub fn text(mut self, content: &str) -> Self {
        self.add(content, None, false);
        self
    }
```


Instead of `add_red` this one:

```rust
    pub fn red(mut self, content: &str) -> Self {
        self.add(content, Some(Color::Red), false);
        self
    }
```


And the sample could become:

```rust

     #[test]
     fn test_hello() {
         colored::control::set_override(true);
         let message = RichText::new().text("Hello ").red("Bob").text("!");
         assert_eq!(message.plain(), "Hello Bob!");
         assert_eq!(message.rich(), "Hello \u{1b}[31mBob\u{1b}[0m!");
     }
```